src/script/sign.cpp

```diff
@@ -392,22 +392,26 @@ SignatureData CombineSignatures(const CScript& scriptPubKey, const BaseSignature
 
 namespace {
 /** Dummy signature checker which accepts all signatures. */
-class DummySignatureChecker : public BaseSignatureChecker
+class DummySignatureChecker final : public BaseSignatureChecker
 {
 public:
-    DummySignatureChecker() {}
-
     bool CheckSig(const std::vector<unsigned char>& scriptSig, const std::vector<unsigned char>& vchPubKey, const CScript& scriptCode, SigVersion sigversion) const override
     {
         return true;
     }
 };
-const DummySignatureChecker dummyChecker;
+const DummySignatureChecker dummy_checker;
+
+class DummySignatureCreator final : public BaseSignatureCreator {
+public:
+    const BaseSignatureChecker& Checker() const override;
+    bool CreateSig(const SigningProvider& provider, std::vector<unsigned char>& vchSig, const CKeyID& keyid, const CScript& scriptCode, SigVersion sigversion) const override final;
```

Comment:
Remove `final` since class definition is final? Or add `final` in `DummySignatureChecker::CheckSig` above?